### PR TITLE
Add headers field to Tempo's remote_write  

### LINKED ISSUE #535: Support specifying additional headers in the remote_write in Tempo config
`remote_write`s  for the Prometheus configuration support specification of additional headers that are passed to the remote writer. Unfortunately this doesn't apply for the `remote_write`s in Tempo's config.  

At work we have a requirement where this would come in very handy. I looked into the code and it seems to be pretty straightforward to add and I could provide a PR.

### PR DESCRIPTION
**Pull request by alexbiehl**
#### PR Description 

This PR adds an optional `headers` field to `remote_write`s in Tempo. Headers specified through that field are passed on to remote writers. 

#### Which issue(s) this PR fixes 

Closes #535 

#### PR Checklist

I will update documentation and CHANGELOG in upcoming commits.

- [x] CHANGELOG updated 
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

**Review by rfratto — APPROVED** [review]
This is great, thanks! Tagging @mapno for a code owner review. (Though we should hold off on merging until you have the changelogs/docs updated and we briefly review those as well)

**Comment by alexbiehl** [comment]
Thanks for confirming this is the right approach @rfratto. I have added a CHANGELOG entry and adjusted the documentation.

File: docs/configuration-reference.md
Comment (rfratto) [inline docs/configuration-reference.md:2020]: Can you note that "authorization" will always be overridden here?

Comment (alexbiehl) [reply docs/configuration-reference.md:2020]: Hm, is it really always or just in presence of basic_auth?

Comment (rfratto) [reply docs/configuration-reference.md:2020]: Ah, that's right, my bad. Let's note that configuring basic auth will override the value for authorization if set in custom headers, then.

**Review by mapno — APPROVED** [review]
LGTM, thanks for the contribution!

**Pull request merged**
